### PR TITLE
Check in/62227/create upcoming appointments component

### PR DESCRIPTION
**Pull request by wullaski**
## Summary

- Adds the upcoming appointments list to the new appointments landing page
- Adds mock api call and data
- Adds Unit tests for components
- Adds Cypress tests for the new list

## Related issue(s)

- [department-of-veterans-affairs/va.gov-team#62227](https://github.com/department-of-veterans-affairs/va.gov-team/issues/62227)

## Testing done

- Unit
- Cypress

## Screenshots

Screen Shot 2023-09-26 at 15 14 21

### CODE REVIEW

**Review by CBonade — CHANGES_REQUESTED** [review]
This branch will need to be updated from main to fix CI before it can be merged. The E2E suite is getting skipped due to the allow list fetching failing. It's failing due to a missing env variable, as this branch is a fair bit behind the main branch and is missing a handful of new features.

File: src/applications/check-in/components/UpcomingAppointments.jsx
Comment (brianseek) [inline src/applications/check-in/components/UpcomingAppointments.jsx:64]: non-translated string here

File: src/applications/check-in/components/UpcomingAppointmentsListItem.jsx
Comment (brianseek) [inline src/applications/check-in/components/UpcomingAppointmentsListItem.jsx:80]: Clinic should also be translated

File: src/applications/check-in/containers/withError.jsx
Comment (brianseek) [inline src/applications/check-in/containers/withError.jsx:28]: add this var to the dependency array

File: src/applications/check-in/hooks/useGetUpcomingAppointmentsData.jsx
Comment (brianseek) [inline src/applications/check-in/hooks/useGetUpcomingAppointmentsData.jsx:46]: missing dependency

File: src/applications/check-in/components/UpcomingAppointmentsList.jsx
Comment (brianseek) [inline src/applications/check-in/components/UpcomingAppointmentsList.jsx:61]: these children need a unique key. you can use <React.Fragment key={uniqueKey}> to do that

File: src/applications/check-in/utils/appointment/index.js
Comment (brianseek) [inline src/applications/check-in/utils/appointment/index.js:174]: These months need to be translatable

File: src/applications/check-in/utils/appointment/index.js
Comment (brianseek) [inline src/applications/check-in/utils/appointment/index.js:188]: these day abbreviations also need to be translatable

File: src/applications/check-in/hooks/useGetUpcomingAppointmentsData.jsx
Comment (brianseek) [inline src/applications/check-in/hooks/useGetUpcomingAppointmentsData.jsx:32]: I think you should be checking if you already have the data before requesting it. Otherwise the page will make a call every time someone goes to it. For example if you click details and then back to list. You could borrow the needsRefresh concept from the other hook or figure out something else. This can also be spun out into another ticket since the focus of this ticket was the display side.

**Review by brianseek — CHANGES_REQUESTED** [review]
This looks great for the most part! A few translation issues and a child key thing. There is also a bigger issue that you can choose to solve here or create a new ticket for.

File: src/applications/check-in/components/UpcomingAppointments.jsx
Comment (wullaski) [reply src/applications/check-in/components/UpcomingAppointments.jsx:64]: UX is working on a solution for this. Thought I would leave as is until that's ready.

File: src/applications/check-in/hooks/useGetUpcomingAppointmentsData.jsx
Comment (wullaski) [reply src/applications/check-in/hooks/useGetUpcomingAppointmentsData.jsx:32]: I created a ticket https://github.com/department-of-veterans-affairs/va.gov-team/issues/66619

File: src/applications/check-in/utils/appointment/index.js
Comment (wullaski) [reply src/applications/check-in/utils/appointment/index.js]: I refactored this so it doesn't rely on the keys for display

File: src/applications/check-in/utils/appointment/index.js
Comment (wullaski) [reply src/applications/check-in/utils/appointment/index.js]: Refactored and removed these

File: src/applications/check-in/components/UpcomingAppointments.jsx
Comment (brianseek) [reply src/applications/check-in/components/UpcomingAppointments.jsx:64]: I would say to add it as a temp key now, so we don't forget it. Also since we have a no hard coded strings check explicitly in our QA steps

File: src/applications/check-in/hooks/useGetUpcomingAppointmentsData.jsx
Comment (brianseek) [reply src/applications/check-in/hooks/useGetUpcomingAppointmentsData.jsx]: this is still missing `setUpcomingAppointmentsData`

**Review by brianseek — APPROVED** [review]
I added two more items but am approving this, so you can merge away after addressing those things. Nice work here!

**Pull request merged**
